src/gbastd/types/math.h

```diff
@@ -0,0 +1,117 @@
+#ifndef MATH.H
```

Comment:
Also math.h I think is a bit to general.(Also this name has already exist in C std, so what can lead to misunderstanding), maybe rename it to 'trigonometry' or 'angles'